### PR TITLE
Improve VNC screenshot test

### PR DESCRIPTION
**Pull request by rmohr**
**What this PR does / why we need it**:

It turns out that eventually the VNC screen ends up with a resolution of 720x400, but the initial seabios screen has a different resolution for a very short amount of time.

Cope with that by retrying until we see the expected resolution.

Example run: https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/batch/pull-kubevirt-e2e-k8s-1.22-sig-compute/1574529294523174912

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: tests/vnc_test.go [hunk 1]
```diff
@@ -246,17 +247,20 @@ var _ = Describe("[rfe_id:127][crit:medium][arm64][vendor:cnv-qe@redhat.com][lev
 		It("should allow creating a VNC screenshot in PNG format", func() {
 			filePath := filepath.Join(GinkgoT().TempDir(), "screenshot.png")
 
-			cmd := clientcmd.NewVirtctlCommand("vnc", "screenshot", "--namespace", vmi.Namespace, "--file", filePath, vmi.Name)
-			Expect(cmd.Execute()).To(Succeed())
+			// Sometimes we can see initially a 640x480 resolution if we connect very early
+			By("gathering screenshots until we are past the first boot screen and see the expected 720x400 resolution")
+			Eventually(func() image.Image {
+				cmd := clientcmd.NewVirtctlCommand("vnc", "screenshot", "--namespace", vmi.Namespace, "--file", filePath, vmi.Name)
```
Comment (xpivarc) [inline tests/vnc_test.go:253]: I did not check and I rather ask, will we override the previous file?

**Review by xpivarc — COMMENTED** [review]
/approve
Thanks for the fast turnaround.

File: tests/vnc_test.go (hunk 1 above)
Comment (rmohr) [reply tests/vnc_test.go:253]: Yes, `os.WriteFile` truncates the file if necessary first.

File: tests/vnc_test.go [hunk 2]
```diff
@@ -299,3 +303,47 @@ func upgradeCheckRoundTripperFromConfig(config *rest.Config, subprotocols []stri
...
+	}
+	return fmt.Sprintf("Expected X: %d, Y: %d to match X: %d, Y: %d", x, y, h.X, h.Y)
+}
+
+func (h ResolutionMatcher) NegatedFailureMessage(actual interface{}) (message string) {
+	x, y, err := imgSize(actual)
+	if err != nil {
+		return err.Error()
+	}
+	return fmt.Sprintf("Expected: %d, Y: %d, to not match X: %d, Y: %d", x, y, h.X, h.Y)
```
[the remark this answers is not in the document]
Comment (rmohr) [reply tests/vnc_test.go]: fixed. :+1:

**Review by brianmcarey — COMMENTED** [review]
/lgtm

**Review by dhiller — APPROVED** [review]
/approve

**Comment by rmohr** [comment]
/override pull-kubevirt-e2e-k8s-1.22-sig-storage

Failure is unrelated, let's get the test stable.

**Pull request merged**